src/Compilers/CSharp/Portable/Binder/RefSafetyAnalysis.cs

```diff
@@ -0,0 +1,896 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal sealed partial class RefSafetyAnalysis : BoundTreeWalkerWithStackGuardWithoutRecursionOnTheLeftOfBinaryOperator
```

Comment:
>RefSafetyAnalysis [](http://example.com/codeflow?start=34&length=17)

Does this visitor throw or silently walk through when encountering nodes that it doesn't handle? An exception would be desirable (override `DefaultVisit` and throw)

Reply:
The visitor should be visiting all nodes, although most nodes simply use the `BoundTreeWalker` implementation to visit child nodes rather than performing some ref safety check. `DefaultVisit()` is not called from `BoundTreeWalker`. Perhaps we could change this implementation in the future to explicitly visit all nodes here rather than relying on the base class, at which time we could throw an exception for unhandled node types.